pkg/core/object/metadata.go

```diff
@@ -0,0 +1,115 @@
+package object
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"slices"
+	"strings"
+
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+)
+
+// SearchResultMerger merges SearchV2 results from several sources. Must not be
+// used concurrently.
+type SearchResultMerger struct {
+	limit, n    uint16
+	more        bool
+	buf         []client.SearchResultItem
+	iBuf, iNext *big.Int
+}
+
+// NewSearchResultMerger constructs SearchResultMerger merging up to limit
+// results.
+func NewSearchResultMerger(limit uint16) *SearchResultMerger {
+	return &SearchResultMerger{limit: limit}
+}
+
+// Add accumulates the next result.
+func (x *SearchResultMerger) Add(withCursor bool, next []client.SearchResultItem) {
+	if len(next) > int(x.limit) {
```

Comment:
i do not understand these panics: a merger is created with some limit and then it is filled with some number of `Add` calls. this number is undefined and is based on shard number. why is it not possible (not expected) to exceed it and why results must be exactly `limit` len?

Reply:
gone